test/setup_bundler_test.rb

```diff
@@ -270,6 +270,38 @@ def test_creates_custom_bundle_with_specified_branch
     end
   end
 
+  def test_install_prerelease_versions_if_experimental_is_true
+    Dir.mktmpdir do |dir|
+      Dir.chdir(dir) do
+        File.write(File.join(dir, "Gemfile"), <<~GEMFILE)
+          source "https://rubygems.org"
+          gem "rdoc"
+        GEMFILE
+
+        capture_subprocess_io do
+          Bundler.with_unbundled_env do
+            # Run bundle install to generate the lockfile
+            system("bundle install")
+
+            # Run the script once to generate a custom bundle
+            run_script
+          end
+        end
+
+        capture_subprocess_io do
+          Object.any_instance.expects(:system).with(
+            bundle_env(".ruby-lsp/Gemfile"),
+            "((bundle check && bundle update ruby-lsp debug --pre) || bundle install) 1>&2",
```

Comment:
Is there a risk of unintentionally using a pre-release version of `debug`?

Reply:
If there's a prerelease version of debug, it will definitely be installed. There's no way to prevent this though - unless we never upgraded debug.

Reply:
Yes, but that'll run `bundle update` twice, which is slow.

Reply:
Yeah, but if it only impacts the experimental flag then it's not so critical.

I assume `bundle update debug --pre ruby-lsp` isn't possible, i.e. `--pre` applies to everything?

Not a blocker anyway, feel free to ship.

Reply:
I think `--pre` applies to everything. To make it run two bundle updates only for the experimental flag will involve complicating the code a little more, since now we're just appending `--pre` at the end.

Let's ship like this for now and revisit in the future if it becomes an issue.